shared/public/version.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+#include "system.h"
+
+namespace ink {
+	constexpr uint32_t VERSION = 0;
```

Comment:
Compile-time constants should be in PascalCase.

Also maybe name this something a bit more specific like `InkBinVersion` perhaps?

https://github.com/brwarner/inkcpp/blob/master/CONTRIBUTING.md#code-style